packages/cactus-plugin-keychain-azure-kv/src/main/typescript/credentials/access-token-credential.ts

```diff
@@ -0,0 +1,60 @@
+import type { Logger, LogLevelDesc } from "@hyperledger/cactus-common";
+import type { AccessToken, TokenCredential } from "@azure/identity";
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { RefreshTokenCredential } from "./refresh-token-credential";
+
+import { LoggerProvider } from "@hyperledger/cactus-common";
+
+export interface IAccessTokenCredentialOptions {
+  readonly logLevel?: Readonly<LogLevelDesc>;
+  readonly accessToken: Readonly<AccessToken>;
+}
+
+/**
+ * An overly simplistic credential implementation that is compatible with the
+ * Azure Javascript SDK.
+ * It doesn't handle refreshing the token, the caller is responsible for updating
+ * the state of it to ensure that the token that it returns hasn't expired.
+ *
+ * Should be useful for local development scenarios for the most part since in
+ * production you would most likely want to go with {@link RefreshTokenCredential}
+ *
+ * @see {@link RefreshTokenCredential}
+ */
+export class AccessTokenCredential implements TokenCredential {
+  public static readonly CLASS_NAME = "AccessTokenCredential";
+
+  private readonly log: Logger;
+
+  public get className(): string {
+    return AccessTokenCredential.CLASS_NAME;
+  }
+
+  constructor(private opts: Readonly<IAccessTokenCredentialOptions>) {
+    const fn = `${this.className}#constructor()`;
+    if (!opts) {
+      throw new Error(`${fn} arg opts cannot be falsy.`);
+    }
+    const { accessToken, logLevel = "WARN" } = opts;
+    if (!accessToken) {
+      throw new Error(`${fn} arg opts.accessToken cannot be falsy.`);
+    }
+    const { expiresOnTimestamp, token } = accessToken;
+    if (typeof token !== "string" || token.trim().length < 1) {
+      throw new Error(`${fn} arg accessToken.token must be non-blank string.`);
+    }
+    if (!Number.isInteger(expiresOnTimestamp)) {
+      throw new Error(`${fn} arg accessToken.expiresOnTimestamp must be int`);
+    }
+
+    this.log = LoggerProvider.getOrCreate({
+      label: this.className,
+      level: logLevel,
+    });
+    this.log.debug("Created instance OK.");
+  }
+  async getToken(): Promise<AccessToken | null> {
+    this.log.debug("ENTER");
```

Comment:
What does this debug statement signify?

Reply:
Just that the execution has entered the method as expected. It might seem trivial, but it helped me a lot during debugging when I had multiple smaller bugs related to how the API server was loading the plugin (or so I thought).
In short, 'ENTER' as in "entering the execution of this method. It's a pattern more frequently seen in Java (I'm old) but I like to adopt it in NodeJS projects too sometimes